static/app/views/replays/searchBar.tsx

```diff
@@ -0,0 +1,144 @@
+import {useEffect} from 'react';
+import assign from 'lodash/assign';
+import flatten from 'lodash/flatten';
+import memoize from 'lodash/memoize';
+import omit from 'lodash/omit';
+
+import {fetchTagValues} from 'sentry/actionCreators/tags';
+import SmartSearchBar from 'sentry/components/smartSearchBar';
+import {NEGATION_OPERATOR, SEARCH_WILDCARD} from 'sentry/constants';
+import {Organization, TagCollection} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {FIELD_TAGS, isAggregateField} from 'sentry/utils/discover/fields';
+import {FieldKind} from 'sentry/utils/fields';
+import useApi from 'sentry/utils/useApi';
+import withTags from 'sentry/utils/withTags';
+
+const SEARCH_SPECIAL_CHARS_REGEXP = new RegExp(
+  `^${NEGATION_OPERATOR}|\\${SEARCH_WILDCARD}`,
+  'g'
+);
+
+const REPLAY_TAGS = [
+  'device.brand',
+  'device.family',
+  'device.name',
+  'dist',
+  'os.build',
+  'os.kernel_version',
+  'platform.name',
+  'release',
+  'sdk.name',
+  'sdk.version',
+  'user.display',
+  'user.email',
+  'user.id',
+  'user.ip',
+  'user.username',
+];
```

Comment:
@cmanallen can you confirm the list of searchable fields?

Reply:
Some fields already exist in the regular API but they are named differently here. IE: ipAddress and the os fields. Should the API align on these or should I just add new search fields for these specifically for replay search?

Reply:
@cmanallen Can we double-check whether it's `user.ipAddress` (per your comment) or `user.ip_address` (api response returns this)


There are some missing fields in this list, which is fine. Listing them for clarity:
- title (idk what's in here, nvm this one)
- errorIds, traceIds, urls
- startedAt, finishedAt
- countUrls
- longestTransaction (gte search here would be cool)
- environment

Reply:
For the search it's ipAddress. If the search doesn't support the param it returns an error response. The reason I was asking is because existing search fields that we have already have descriptions and everything configured. So for these we'll need to create new ones specifically for replay queries (like I did with a few of the replay specific ones)